syncapi/synctypes/clientevent.go

```diff
@@ -78,63 +80,62 @@ func ToClientEvents(serverEvs []gomatrixserverlib.PDU, format ClientEventFormat,
 		if se == nil {
 			continue // TODO: shouldn't happen?
 		}
-		if format == FormatSyncFederation {
-			evs = append(evs, ToClientEvent(se, format, string(se.SenderID()), se.StateKey(), spec.RawJSON(se.Unsigned())))
+		ev, err := ToClientEvent(se, format, userIDForSender)
+		if err != nil {
+			logrus.Errorf("Failed converting event to ClientEvent: %s", err.Error())
```

Comment:
Since we're continuing, maybe warn?
`logrus.WithError(err).Warn("Failed to convert event to ClientEvent")`?